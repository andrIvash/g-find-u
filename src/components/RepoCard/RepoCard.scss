
@import '../../styles/abstracts/functions';
@import '../../styles/abstracts/breakpoints';
// .repoCards {
//     list-style: none;
//     margin: 0;
//     padding: 0;
// }

// .repoCards__item {
//     padding: 20px 10px;
//     border-bottom: 1px #e1e4e8 solid;
//     cursor: pointer;
// }
  
// .repoCards__item:focus, .repoCards__item:hover {
//     box-shadow: -1px -1px 9px 1px rgba(0,0,0,0.08);
//     outline: none;
// }
$star-width: rem(40);
$margin: rem(20);
$padding: rem(10);
$logo-size: rem(10);

.repoCard {
    .repoCard__header {
        display: flex;
        margin-bottom: $padding;
    }

    .repoCard__title {
        line-height: 1;
    }

    .repoCard__headerLeft {
        flex: 1;
    }

    .repoCard__headerRight {
        display: flex;
        justify-content: center;
        align-items: flex-start;
        width: $star-width;

        a {
            width: 100%;
            display: flex;
            justify-content: center;
            align-items: center;
        }

        svg path {
            fill: var(--app-color);
        }
    }

    .repoCard__stars {
        margin-right: calc($padding / 2);
        color: var(--app-color);
        font-weight: bold;
    }

    .repoCard__link, .repoCard__link:active, .repoCard__link:hover, .repoCard__link:visited{
        color: var(--app-color);
        text-decoration: none;
        position: relative;
    }


    .repoCard__desq {
        color: var(--app-color);
    }

    .repoCard__footer {
        display: flex;
        align-items: center;
        font-size: var(--app-font-comment-size);
    }

    .repoCard__lang-logo {
        display: inline-block;
        width: $logo-size;
        height: $logo-size;
        height: $logo-size;
        border-radius: 50%;
        margin-right: calc($padding / 2);
    }

    .repoCard__lang {
        margin-right: calc($padding / 2);
        font-weight: bold;
    }
}

@media screen {
    @include break-min-max(rem(425), rem(1100)) {
        .repoCard {
            .repoCard__header {
                margin-bottom: $padding;
            }

            .repoCard__title {
                margin-bottom: calc($padding / 2);
            }

            .repoCard__footer {
                font-size: var(--app-font-comment-size-medium);
            }
        }
    }

    @include break-min-max(rem(320), rem(500)) {
        .repoCard {
            .repoCard__header {
                margin-bottom: calc($padding / 2);
            }

            .repoCard__title {
                margin-bottom: calc($padding / 2);
            }

            .repoCard__footer {
                font-size: var(--app-font-comment-size-small);
            }
        }
    }
}

