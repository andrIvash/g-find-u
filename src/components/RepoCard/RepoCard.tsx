import React from 'react';
import { Props, defaultLangColor } from './index';

import './RepoCard.scss';

export const RepoCard = (props: Props) => {
    return (
        <section className='repoCard'>
            <header className='repoCard__header'>
                <div className='repoCard__headerLeft'>
                    <h3 className='repoCard__title'>
                        <a 
                            className='repoCard__link'
                            target='_blank'
                            rel='noreferrer'
                            href={`${props.repoLink ? props.repoLink : "#"}`}>
                                {props.title}
                        </a>
                    </h3>
                    {props.isFork && <div className='repoCard__fork'>   
                        {`it's Fork`}
                    </div>}
                </div>
                <div className='repoCard__headerRight'>
                    <a className='visually-hidden' href="https://api.github.com/repos/andrIvas/excercises/stargazers">
                        <span className='repoCard__stars'>{props.stars ? props.stars : 0}</span>
                        <svg className="octicon octicon-star" aria-label="star" height="16" role="img" version="1.1" viewBox="0 0 14 16" width="14">
                            <path fillRule="evenodd" d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z"></path>
                        </svg>
                    </a>
                </div>
            </header>
            <main className='repoCard__main'>
                <p className='repoCard__desq'>{props.desc}</p>
            </main>
            <footer className='repoCard__footer'>
                <span
                    className='repoCard__lang-logo'
                    style={{ backgroundColor: props.logoColor ? props.logoColor : defaultLangColor }}></span>
                <span className='repoCard__lang'>{props.lang}</span>
                <span className="repoCard__time">Updated: {props.time}</span>
            </footer>
      </section>
    )
}