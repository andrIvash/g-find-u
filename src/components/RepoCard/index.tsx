import { RepoCard } from './RepoCard';

export const defaultLangColor = '#f1e05';

export interface Props {
   title: string,
   repoLink: string,
   isFork: boolean,
   stars: number,
   desc: string,
   logoColor: string,
   lang: string,
   time: string
}

export default RepoCard;