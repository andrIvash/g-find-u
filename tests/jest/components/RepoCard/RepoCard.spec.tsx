import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import RepoCard from '../../../../src/components/RepoCard';
import { repoCardMockData } from  '../../../mocks/repoCardMockData';

describe('RepoCard Component', () => {
    it('renders without crashing', () => {
        render(<RepoCard {...repoCardMockData} />);
        expect(screen.getByText('Test Repo')).toBeInTheDocument();
    });

    it('displays correct stars count', () => {
        render(<RepoCard {...repoCardMockData} />);
        expect(screen.getByText('10')).toBeInTheDocument();
    });

    it('displays fork indication when isFork is true', () => {
        render(<RepoCard {...repoCardMockData} isFork={true} />);
        expect(screen.getByText("it's Fork")).toBeInTheDocument();
    });

    it('does not display fork indication when isFork is false', () => {
        render(<RepoCard {...repoCardMockData} />);
        expect(screen.queryByText("it's Fork")).toBeNull();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<RepoCard {...repoCardMockData} />);
        expect(asFragment()).toMatchSnapshot();
    });
});