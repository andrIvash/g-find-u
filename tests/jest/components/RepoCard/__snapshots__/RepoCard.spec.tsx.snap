// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCard Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="repoCard"
  >
    <header
      class="repoCard__header"
    >
      <div
        class="repoCard__headerLeft"
      >
        <h3
          class="repoCard__title"
        >
          <a
            class="repoCard__link"
            href="https://github.com/test/repo"
            rel="noreferrer"
            target="_blank"
          >
            Test Repo
          </a>
        </h3>
      </div>
      <div
        class="repoCard__headerRight"
      >
        <a
          class="visually-hidden"
          href="https://api.github.com/repos/andrIvas/excercises/stargazers"
        >
          <span
            class="repoCard__stars"
          >
            10
          </span>
          <svg
            aria-label="star"
            class="octicon octicon-star"
            height="16"
            role="img"
            version="1.1"
            viewBox="0 0 14 16"
            width="14"
          >
            <path
              d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </div>
    </header>
    <main
      class="repoCard__main"
    >
      <p
        class="repoCard__desq"
      >
        Test Description
      </p>
    </main>
    <footer
      class="repoCard__footer"
    >
      <span
        class="repoCard__lang-logo"
        style="background-color: rgb(0, 0, 0);"
      />
      <span
        class="repoCard__lang"
      >
        JavaScript
      </span>
      <span
        class="repoCard__time"
      >
        Updated: 1 day ago
      </span>
    </footer>
  </section>
</DocumentFragment>
`;
